finality-aleph/src/network/service.rs

```diff
@@ -145,24 +173,44 @@ impl<N: Network, D: Data> Service<N, D> {
             }
             Event::NotificationStreamOpened {
                 remote, protocol, ..
-            } => {
-                if protocol == ALEPH_PROTOCOL_NAME {
+            } => match protocol.as_ref().try_into() {
+                Ok(Protocol::Generic) => {
                     trace!(target: "aleph-network", "NotificationStreamOpened event for peer {:?}", remote);
                     let (tx, rx) = mpsc::channel(PEER_BUFFER_SIZE);
                     self.spawn_handle.spawn(
                         "aleph/network/peer_sender",
                         None,
                         self.peer_sender(remote.into(), rx),
                     );
-                    self.connected_peers.insert(remote.into());
-                    self.peer_senders.insert(remote.into(), tx);
+                    self.generic_connected_peers.insert(remote.into());
+                    self.generic_peer_senders.insert(remote.into(), tx);
                 }
-            }
+                Ok(Protocol::Validator) => {
+                    trace!(target: "aleph-network", "NotificationStreamOpened event for peer {:?}", remote);
+                    let (tx, rx) = mpsc::channel(PEER_BUFFER_SIZE);
+                    self.spawn_handle.spawn(
+                        "aleph/network/peer_sender",
+                        None,
+                        self.peer_sender(remote.into(), rx),
+                    );
+                    self.validator_connected_peers.insert(remote.into());
+                    self.validator_peer_senders.insert(remote.into(), tx);
+                }
+                Err(_) => {}
+            },
             Event::NotificationStreamClosed { remote, protocol } => {
-                if protocol == ALEPH_PROTOCOL_NAME {
-                    trace!(target: "aleph-network", "NotificationStreamClosed event for peer {:?}", remote);
-                    self.connected_peers.remove(&remote.into());
-                    self.peer_senders.remove(&remote.into());
+                match protocol.as_ref().try_into() {
+                    Ok(Protocol::Generic) => {
+                        trace!(target: "aleph-network", "NotificationStreamClosed event for peer {:?}", remote);
+                        self.generic_connected_peers.remove(&remote.into());
+                        self.generic_peer_senders.remove(&remote.into());
+                    }
+                    Ok(Protocol::Validator) => {
+                        trace!(target: "aleph-network", "NotificationStreamClosed event for peer {:?}", remote);
+                        self.validator_connected_peers.remove(&remote.into());
+                        self.validator_peer_senders.remove(&remote.into());
+                    }
+                    Err(_) => {}
```

Comment:
Same as in the above thread started by Marcin – a comment about why we don't care about other protocols would be useful.

Reply:
Done